src/service/resources-impl.js

```diff
@@ -1727,27 +1814,53 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
+  unlayout() {
     if (this.state_ == ResourceState_.NOT_BUILT) {
       return;
     }
     if (this.isInViewport()) {
       this.setInViewport(false);
     }
-    if (this.element.documentInactiveCallback()) {
+    if (this.element.unlayoutCallback()) {
       this.state_ = ResourceState_.NOT_LAID_OUT;
     }
   }
 
   /**
-   * Called when a previously visible element has become invisible.
+   * Calls element's pauseCallback callback.
+   */
+  pause() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    this.element.pauseCallback();
+  }
+
+  /**
+   * TODO(jridgewell): This should eventually call layoutCallback
```

Comment:
So, I really don't think there's `layout` method here - I think we need to remove it to avoid bugs and/or confusion.